src/mpid/ch4/src/ch4_types.h

```diff
@@ -278,6 +278,17 @@ typedef struct {
 #define MPIDIU_THREAD_MPIDIG_GLOBAL_MUTEX MPIDI_global.m[3]
 #define MAX_CH4_MUTEXES 4
 
+/* VCI */
+typedef struct MPIDI_vci {
+    MPID_Thread_mutex_t lock;   /* lock to protect this VCI */
+    MPIR_Request *lw_req;       /* pre-allocated completed request for this VCI */
+    MPIDI_workq_t workqueue;    /* WorkQ for this VCI */
+} MPIDI_vci_t;
+
+#define MPIDI_VCI_FALLBACK 0
+
+#define MPIDI_VCI(i) MPIDI_global.vci[i]
```

Comment:
The macro hardly saves any typing but adds one layer of indirection. I would recommend do without the macro.